modules/mono/glue/GodotSharp/GodotSharp/Core/Dictionary.cs

```diff
@@ -449,6 +449,11 @@ private KeyValuePair<Variant, Variant> GetKeyValuePair(int index)
                 Variant.CreateTakingOwnershipOfDisposableValue(value));
         }
 
+        public KeyValuePair<Variant, Variant> GetAt(int index)
+        {
+            return GetKeyValuePair(index);
+        }
```

Comment:
Added documentation and exceptions.

```suggestion
        /// <summary>
        /// Returns the entry at the given <paramref name="index"/>.
        /// </summary>
        /// <exception cref="ArgumentOutOfRangeException">
        /// <paramref name="index"/> is less than 0 or greater than the dictionary's size.
        /// </exception>
        /// <param name="index">The index of the entry to retrieve.</param>
        /// <returns>The entry for the given <paramref name="index"/>.</returns>
        public KeyValuePair<Variant, Variant> GetAt(int index)
        {
            if (index < 0 || index >= Count)
                throw new ArgumentOutOfRangeException(nameof(index));

            return GetKeyValuePair(index);
        }
```

Reply:
The GDScript version does not throw an error for out of bounds, but returns null. Is it fine for C# to have different behavior?

Reply:
C# collections usually throw an exception when the index is out of bounds and we try to be consistent with the BCL.

Reply:
Sounds good!